lib/activerecord-import/adapters/mysql_adapter.rb

```diff
@@ -113,11 +114,27 @@ def sql_for_on_duplicate_key_update_as_hash( table_name, hsh ) # :nodoc:
       qc2 = quote_column_name( column2 )
       "#{table_name}.#{qc1}=VALUES( #{qc2} )"
     end
+    increment_locking_column!(results, table_name)
     results.join( ',')
   end
 
   # Return true if the statement is a duplicate key record error
   def duplicate_key_update_error?(exception) # :nodoc:
     exception.is_a?(ActiveRecord::StatementInvalid) && exception.to_s.include?('Duplicate entry')
   end
+
+  def increment_locking_column!(results, table_name)
+    model = model(table_name)
+    if locking_column?(model)
+      results << "#{table_name}.`#{model.locking_column}`=`#{model.locking_column}`+1"
+    end
+  end
+
+  def model(table_name)
+    ActiveRecord::Base.descendants.find { |t| "`#{t.table_name}`" == table_name }
```

Comment:
This complexity worries me a little bit. I'm wondering if maybe it would be better if the locking column was set in the options hash and then passed down like the primary key.

Reply:
🤔 Should I modify this code?

Reply:
Be sure to validate that model has locking_column.
In order to verify, it is necessary to obtain a list of columns held by the model, and it is necessary to find model from the table name.
Because I think it is difficult to delete this method...
If I know the model, I think locking_column is useful because it knows locking_column even if it is not specified by option.
How about that?

Reply:
What I meant was that we could insert the lock_version value into the options hash back in the initial `import_helper` method, similar to how `primary_key` is handled.  If the value is nil, we'll know they table doesn't have a lock column. The trick would be to make sure it gets passed down to these `on_duplicate_key_update` methods.

See: https://github.com/zdennis/activerecord-import/blob/master/lib/activerecord-import/import.rb#L481